app/src/main/java/org/hwyl/sexytopo/control/activity/SexyTopoActivity.java

```diff
@@ -985,6 +983,15 @@ public void onClick(DialogInterface dialog, int id) {
     }
 
 
+    private void toggleDebugMode() {
+        debugMode = !debugMode;
+        dataManager.broadcastSurveyUpdated();
```

Comment:
I really just want to call invalidate on the canvas so that you instantly see the path points but this will call the sync method hence how I noticed how long it would take to refresh. It's fine as long as the table has never been loaded though so I just don't do that. Worth broadcasting an invalidate style message instead?

Reply:
Ahah - are you saying that the table view gets updated even if it's not currently active? That does make sense - it would explain a couple of bug reports about things hanging when updating data. And since I rarely use it that might explain why I didn't see the issue.

I assumed it wouldn't do anything f it's not currently running. (Well actually I suppose I hadn't thought about it.)

I don't think it's worth a separate broadcast option. There's probably a command to tell the activity to redraw itself.

However, it's a bit useless having the table update itself even when you can't see it. I think we need to add a check at the start of the sync methods to not bother if the activity is not Resumed and then sync in `onResume()`.

Reply:
Yep to repro:
 * Load that survey you sent me
 * Load table view
 * Load plan view
 * Pan away from the main survey (to discount draw time)
 * Draw a small sketch
 * Toggle debug mode